api/end_device.proto

```diff
@@ -860,4 +860,7 @@ message ConvertEndDeviceTemplateRequest {
   string format_id = 1 [(validate.rules).string = {pattern: "^[a-z0-9](?:[-]?[a-z0-9]){2,}$", max_len: 36}];
   // Data to convert.
   bytes data = 2;
+  
+  // Mandatory when loading the device profile from the Device Repository.
+  ApplicationIdentifiers application_ids = 3;
```

Comment:
Yeah so let's get rid of this, see https://github.com/TheThingsNetwork/lorawan-stack/issues/5512#issuecomment-1189511262

Reply:
@johanstokking was working on deprecating the appID on DR services and updating the auth validation in this PR but it easily became its own thing, meaning that it impacts quite a few tests. So I ended up making this PR a draft and  will create a separate PR with those changes. That should make it easier to review.